packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.tsx

```diff
@@ -204,12 +220,39 @@ const CreateReceivablesBase = (props: InvoiceDetailsCreateProps) => {
                   onCurrencyChanged={setActualCurrency}
                 />
                 <PaymentSection disabled={createReceivable.isPending} />
-                <ReminderSection disabled={createReceivable.isPending} />
+                <ReminderSection
+                  disabled={createReceivable.isPending}
+                  onUpdateOverdueReminder={onEditOverdueReminder}
+                  onUpdatePaymentReminder={onEditPaymentReminder}
+                  onCreateReminder={onCreateReminder}
+                />
               </Stack>
             </Stack>
           </form>
         </FormProvider>
       </DialogContent>
+
+      <CreateInvoiceReminderDialog
+        open={createReminderDialog.open}
+        reminderType={createReminderDialog.reminderType}
+        onClose={closeCreateReminderDialog}
+        onCreate={({ reminderId, reminderType }) => {
+          if (reminderType === 'payment') {
+            setValue('payment_reminder_id', reminderId);
+          } else if (reminderType === 'overdue') {
+            setValue('overdue_reminder_id', reminderId);
```

Comment:
To maintain clean JSX logic, you can refactor the code to move the logic into a separate function

Reply:
Extracting the logic into a separate function and creating additional callbacks increases the number of necessary types that will inevitably be added to the new function. This also, as I mentioned earlier, removes the logic from its context. In my opinion, this is a flawed and counterproductive practice.

Reply:
The benefits of refactoring code to extract logic into separate functions generally outweigh the drawbacks, even wrapping that in useCallback. It promotes clean, modular, and maintainable code. At the end of the day, it's all about developer preference ...